CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();
+    DoSomerthingLater();
+}
+```
+
+### Remove unused or redudant namespaces
+
+### Use single line and lambda getters
+
+```csharp
+// Good
+public string Name { get; private set; }
+
+// Good
+public string UppercaseName => Name.toUpperCase(); 
+```
+
+### Don't use braces for just one line
+
+```csharp
+// Bad
+if (payment == null)
+{
+    return "A payment is required.";
+}
+
+// Good
+if (payment == null)
+    return "A payment is required.";
+```
+
+### Use spaces in array/object initializers
+
+```csharp
+// Bad
+var names = new string[] {"Marie","John","Paul"};
+
+// Bad
+var names = new string[] {"Marie", "John", "Paul"};
+
+// Good
+var names = new string[] { "Marie", "John", "Paul" };
+```
+
+### Use spaces between operators
+
+```csharp
+// Bad
+var result = 4/2;
+
+// Bad
+var result = 4/2+6;
+
+// Good
+var result = 4 / 2;
+
+// Good
+var result = (4 / 2) + 6;

```

Comment:
I'll check what you did, I'm not sure I have a good way of explaining that. Basically, we want to avoid "weird-looking spacing".

Good:

```csharp
var result = (4 / 2) + 6;
var element = myArray[result];
```

Bad:

```csharp
var result =(4 / 2) + 6; // missing space to the left of opening parenthesis
var result = ( 4 / 2) + 6; // extra space to the right of opening parenthesis
var result = (4 / 2 ) + 6; // extra space to the left of closing parenthesis
var result = (4 / 2)+ 6; // missing space to the right of closing parenthesis

var element = myArray [result] + 1; // extra space to the left of opening bracket
var element = myArray[ result] + 1; // extra space to the right of opening bracket
...
```

You get the idea hehe, there are lots of combinations, and I don't know really how to word them all together.